### ISSUE TITLE
Drop toJSON() in models...maybe?

### ISSUE THREAD

**Issue by mattiamanzati** [mobxjs/mobx-state-tree#69]
I know it's neat, but it breaks typings... :/
See this example:
(running on VSCode insider with noImplicitThis at true)

```typescript
import {Type, types as t} from 'mobx-state-tree'

interface IIdentity{
    id: string
    token: string
}
const Identity = t.model({
    id: t.primitive,
    token: t.primitive
})

const AuthService = t.model({
    user: t.maybe(Identity),
    setLoggedUser(user: IIdentity){
        this.user = user
    }
})
```

**Comment by mweststrate** [comment]
Just tried, doesn't give an error with 0.2.2 for me

**Comment by mweststrate** [comment]
Fixed in #73 and will be part of next release
